sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -682,6 +682,8 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                         return makeBinaryExpressions(node, type, DBSPOpcode.CONCAT, ops);
                     case "array":
                         return this.compileFunction(call, node, type, ops, 0);
+                    case "safe_add":
+                        return makeBinaryExpression(node, type, DBSPOpcode.SAFE_ADD, ops);
```

Comment:
so here instead of makeBinaryExpression use perhaps compilePolymorphicFunction

Reply:
Using `compilePolymorphicFunction` would mean that both the parameters passed in to `safe_add` would need to be the same resulting in the likes of `safe_add(i64, i32)` or, `safe_add(i64, Decimal)` not working. 
And to work around it, we'd need to manually add those implementations, I think.